firmware/Core/Inc/Logs.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+/**
+ * This module holds the interfaces to all data logging (either through Wifi or flash).
+ * Author(s): Elaine
+ */
+
+#include "Common.h"
+
+/***********************************************************************************************************************
+ * Definitions
+ **********************************************************************************************************************/
+
+
+typedef struct
+{
+    float average_temp;
+    float min_temp;
+    float max_temp;
+
+}LogData_t;
+
+/***********************************************************************************************************************
+ * Prototypes
+ **********************************************************************************************************************/
+
+/**
+* Initializes internal parameters.
+* Should be called exactly once before anything is attempted to be done with this module.
+* Failing to call this function, or calling it more than once, voids all guarantees made by this module.
+*/
+void Logs_Init(void);
+
+/**
+* Logs data from sensors, passes to control module.
+* @param[in]        LogData          struct containing the average, min and max temperature
+* @return           LOG_SUCCEEDED or LOG_FAILED
+*/
+int Logs_LogData(const LogData_t *LogData);
+bool Logs_WifiData(const LogData_t *LogData);
+bool Logs_FlashData(const LogData_t *LogData);
```

Comment:
Also need something to log to LCD right ? You should talk to Jacob about how to interact with his lower level LCD driver.

Reply:
oops forgot about that